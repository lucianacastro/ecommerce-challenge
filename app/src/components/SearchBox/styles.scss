@import '../../scss/variables.scss';
.search-box {
    display: flex;
    width: 100%;
    margin-left: 25px;
    &__input {
        width: calc(100% - 40px);
        padding-left: 10px;
        background-color: $white;
        border: none;
        border-radius: 2px 0 0 2px;
        box-sizing: border-box;
        font-size: 16px;
        color: $dark;
        outline: none;
        &::placeholder {
            color: #e6e6e6;
        }
    }
    &__icon {
        width: 18px;
        height: 18px;
        background-image: url('/static/ic_Search@2x.png');
        background-size: cover;
    }
    &__button {
        display: flex;
        justify-content: center;
        align-items: center;
        width: 42px;
        height: 38px;
        padding: 0;
        border: none;
        border-radius: 0 2px 2px 0;
        background-color: $lightGrey;
        outline: none;
        cursor: pointer;
    }
}