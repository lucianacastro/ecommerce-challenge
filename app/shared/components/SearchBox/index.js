import React from 'react';

import './styles.scss';

export default class SearchBox extends React.Component {
    render () {
        return (
            <div className='search-box'>
                <input className='search-box__input' placeholder='Nunca dejes de buscar' />
                <button className='search-box__button' >
                    <div className='search-box__icon' />
                </button>
            </div>
        );
    }
}