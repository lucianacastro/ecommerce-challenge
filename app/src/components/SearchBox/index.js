import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import Router from 'next/router';

import './styles.scss';

export default class SearchBox extends React.Component {

    static displayName = 'SearchBox';

    static propTypes = {
        searchText: PropTypes.string,
    }
    
    state = {
        textInput: this.props.searchText,
    };

    updateTextInput = (event) => {
        const { target } = event;
        if (target) {
            this.setState({ textInput: target.value });
        }
    }

    componentDidMount(){
        this.searchInput.focus(); 
    }

    render () {
        const { textInput = '' } = this.state;
        return (
            <form className='search-box' action='/items' method='GET'>
                <input
                    name="search"
                    className='search-box__input'
                    placeholder='Nunca dejes de buscar'
                    type='text'
                    ref={(input) => { this.searchInput = input; }}
                    value={textInput}
                    onChange={this.updateTextInput}
                />
                <button className='search-box__button' type='submit' >
                    <div className='search-box__icon' />
                </button>
            </form>
        );
    }
}