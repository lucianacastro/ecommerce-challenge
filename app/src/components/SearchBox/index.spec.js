import SearchBox from '.';
import React from 'react';
import { mount, shallow } from 'enzyme';

describe('SearchBox', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(<SearchBox searchText="iphone"/>);
    });

    it('should map snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have a form', () => {
        expect(wrapper.find('form')).toExist();
    });

    it('should have the form action set to "/items"', () => {
        expect(wrapper.find('form')).toHaveProp('action', '/items');
    });

    it('should have the form method set to "GET"', () => {
        expect(wrapper.find('form')).toHaveProp('method', 'GET');
    });

    it('should have a button with type submit', () => {
        expect(wrapper.find('form').find('button')).toHaveProp('type', 'submit');
    });
});