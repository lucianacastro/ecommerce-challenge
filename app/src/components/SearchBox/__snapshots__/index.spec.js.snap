// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should map snapshot 1`] = `
<SearchBox
  searchText="iphone"
>
  <form
    action="/items"
    className="search-box"
    method="GET"
  >
    <input
      className="search-box__input"
      name="search"
      onChange={[Function]}
      placeholder="Nunca dejes de buscar"
      type="text"
      value="iphone"
    />
    <button
      className="search-box__button"
      type="submit"
    >
      <div
        className="search-box__icon"
      />
    </button>
  </form>
</SearchBox>
`;
